src/core/cgroup.c

```diff
@@ -484,7 +486,26 @@ static uint64_t cgroup_context_cpu_shares(CGroupContext *c, ManagerState state)
                 return CGROUP_CPU_SHARES_DEFAULT;
 }
 
-static void cgroup_apply_unified_cpu_config(Unit *u, uint64_t weight, uint64_t quota) {
+static void cgroup_cpu_adjust_quota_period(Unit *u, uint64_t *quota, uint64_t *period) {
+        /* quota -> in as a % of a second, out as an actual USEC value relative to period. */
+
+        if (*quota == USEC_INFINITY)
+                /* Always use default period for infinity quota. */
+                *period = CGROUP_CPU_QUOTA_DEFAULT_PERIOD_USEC;
+        else {
+                if (*period == 0)
```

Comment:
as mentioned elsewhere we should really use == USEC_INFINITY as marker for "default period" not zero, for the only reason that we tend to standardize using a type's max value for marking "undefined"